### PR TITLE
chore(deps): bump golang-ci version to support go 1.21

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Support go v1.21.

Will fix CI for: https://github.com/celestiaorg/celestia-app/pull/2540

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
             go.sum
       - uses: golangci/golangci-lint-action@v3.7.0
         with:
-          version: v1.52.2
+          version: v1.54.2
```
Comment (rootulp) [inline .github/workflows/lint.yml:30]: Confirming this is the latest release: https://github.com/golangci/golangci-lint/releases/tag/v1.54.2

https://github.com/golangci/golangci-lint/releases/tag/v1.54.0 introduced Go 1.21 support.

**Review by rootulp — APPROVED** [review]
it would be nice if dependabot auto-bumped this for us but I think the current Dependabot config doesn't look at this version

**Comment by SweeXordious** [comment]
I guess it should:
https://github.com/celestiaorg/celestia-app/blob/cc8a3a834011d30fc87718124eab32f5d69ab5b5/.github/dependabot.yml#L18-L24

that's weird

**Comment by rootulp** [comment]
I think that config will just monitor the Github action version which in this case is https://github.com/golangci/golangci-lint-action/releases/tag/v3.7.0

**Comment by SweeXordious** [comment]
aaaah makes sense :+1:

**Review by liamsi — APPROVED** [review]

**Pull request merged**